src/java/org/apache/cassandra/index/sai/disk/io/IndexInputReader.java

```diff
@@ -33,6 +33,9 @@
  */
 public class IndexInputReader extends IndexInput
 {
+    // the byte order of `input`'s native readX operations doesn't matter,
+    // because we only use `readFully` and `readByte` methods. IndexInput calls these
+    // (via DataInput) with methods that enforce LittleEndian-ness.
```

Comment:
Another fun way to do this that adds some navigability would be...

```
/**
 * The byte order of {@link RandomAccessReader}'s native read operations doesn't matter,
 * because we only use {@link RandomAccessReader#readFully} and {@link RandomAccessReader#readByte} methods.
 * IndexInput calls these (via {@link DataInput}) with methods that enforce LittleEndian-ness.
 */
```

Reply:
I could go either way on this, in general I prefer to keep implementation details as a comment instead of javadoc

Reply:
Yeah, most people will never read offline JavaDoc for this stuff, so I like to be able to click through to things in-IDE :)